api/task_manager_test.cc

```diff
@@ -0,0 +1,104 @@
+        sharded<task_manager>& tms = ctx.tm;
+        auto it = req->query_parameters.find("task_id");
+        task_id id = it != req->query_parameters.end() ? task_id{utils::UUID{it->second}} : task_id::create_null_id();
+        it = req->query_parameters.find("shard");
+        unsigned shard = it != req->query_parameters.end() ? boost::lexical_cast<unsigned>(it->second) : 0;
+        it = req->query_parameters.find("keyspace");
+        std::string keyspace = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("table");
+        std::string table = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("type");
+        std::string type = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("entity");
+        std::string entity = it != req->query_parameters.end() ? it->second : "";
+        it = req->query_parameters.find("parent_id");
+        task_id parent_id = it != req->query_parameters.end() ? task_id{utils::UUID{it->second}} : task_id::create_null_id();
+
+        auto module = tms.local().find_module("test");
+        id = co_await module->make_task<test_task_impl>(shard, id, keyspace, table, type, entity, parent_id);
+        co_await tms.invoke_on(shard, [id] (task_manager& tm) {
+            auto it = tm.get_all_tasks().find(id);
+            if (it != tm.get_all_tasks().end()) {
+                it->second->start();
+            }
+        });
+        co_return id.to_sstring();
+    });
+
+    tmt::unregister_test_task.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        task_id id = task_id{utils::UUID{req->query_parameters["task_id"]}};
+        co_await task_manager::invoke_on_task(ctx.tm, id, [] (task_manager::task_ptr task) -> future<> {
+            task->finish();
+            task->unregister_task();
+            return make_ready_future<>();
+        });
+        co_return json_void();
+    });
+
+    tmt::finish_test_task.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        task_id id = task_id{utils::UUID{req->param["task_id"]}};
+        auto it = req->query_parameters.find("error");
+        bool fail = it != req->query_parameters.end();
+        std::string error = fail ? it->second : "";
+
+        co_await task_manager::invoke_on_task(ctx.tm, id, [fail, error = std::move(error)] (task_manager::task_ptr task) -> future<> {
+            if (fail) {
+                task->finish_failed(std::make_exception_ptr(std::runtime_error(error)));
+            } else {
+                task->finish();
+            }
+            return make_ready_future<>();
+        });
+        co_return json_void();
+    });
+
+    tmt::get_and_update_ttl.set(r, [&ctx, cfg] (std::unique_ptr<request> req) -> future<json::json_return_type> {
```

Comment:
I don't it used anywhere.
Did I miss anything?

Reply:
It is used here:
https://github.com/scylladb/scylladb/blob/573245e2ca12cf286c104cbd80de8bdda0f10658/test/rest_api/rest_util.py#L73-L82
And then it's called in three tests in `test/rest_api/test_task_manager.py` in lines 82, 98, and 142.

Reply:
(TODO: Unnecessarily formatted strings)